lua/nvim-treesitter/textobjects.lua

```diff
@@ -0,0 +1,95 @@
+local api = vim.api
+local ts = vim.treesitter
+
+local configs = require "nvim-treesitter.configs"
+local parsers = require "nvim-treesitter.parsers"
+local queries = require'nvim-treesitter.query'
+local locals = require'nvim-treesitter.locals'
+local ts_utils = require'nvim-treesitter.ts_utils'
+
+local M = {}
+
+function M.select_textobj(query_string)
+  local bufnr = vim.api.nvim_get_current_buf()
+  local ft = api.nvim_buf_get_option(bufnr, "ft")
+  if not ft then return end
+  local lang = parsers.ft_to_lang(ft)
+
+  local row, col = unpack(vim.api.nvim_win_get_cursor(0))
+  row = row - 1
+
+  local matches = {}
+
+  if string.match(query_string, '^@.*') then
+    matches = locals.get_capture_matches(bufnr, query_string, 'textobjects')
```

Comment:
As discussed, this will have to be moved to `query` instead of `locals`.

Reply:
https://github.com/nvim-treesitter/nvim-treesitter/pull/161

You're right. But we should try not to change to much stuff at once. Better let users test that feature before touching more modules. This PR already affect `textobjects` and `incremental_selection`